firmware/arduino/chessboard/movementFunctions.ino

```diff
@@ -1,72 +1,259 @@
+  int8_t deltaX, deltaY, subDeltaX, subDeltaY, absDeltaX, absDeltaY;
+  uint8_t i, statusCodeResult;
+  uint8_t curCol, curRow;
+  uint8_t numPoints = 0;
+
+  // Find the signed difference between the finial and initial points
+  deltaX = endCol - startCol;
+  deltaY = endRow - startRow;
+
+  // We only need the magnitude of deltaX and deltaY after this point 
+  absDeltaX = abs(deltaX);
+  absDeltaY = abs(deltaY);
+
+  // Calculate the sub-distances for X and Y movement. They can be overridden as necessary
+  subDeltaX = deltaX - (2 * diagDirX);
+  subDeltaY = deltaY - (2 * diagDirY);
+
+  // Calculate the direction of the diagonals
+  diagDirX = (deltaX > 0) ? 1 : -1;
+  diagDirY = (deltaY > 0) ? 1 : -1;
+
+  // If target point is equal to the start point
+  if (deltaX == 0  &&  deltaY == 0)
     return true;
+    
+  curCol = currPositionX / (stepsPerUnitSpace * 8);
+  curRow = currPositionY / (stepsPerUnitSpace * 8);
+
+  // Make initial move to first position. Move diagonally since it's faster.
+  if (!moveDirect(curCol, curRow, startCol, startRow))
+    return false;
+
+  // Calculate a list of up to 4 new points (including the start point) 
+  // We use previous points to calculate subsequent points, since they're on the same path
+  // Initial position
+  cols[0] = startCol;
+  rows[0] = startRow;
+
+  // Case where we should call moveDirect, since pathing can be simplified to that
+  if (absDeltaX <= 2  &&  absDeltaY <= 2)
+    return moveDirect(startCol, startRow, endCol, endRow);
+
+  // Because of the checks above, we know absDeltaX and absDeltaY can't both be 0 at the same time
+  // Case where we have a strictly vertical or horizontal movement along edges
+  if (absDeltaX == 0  ||  absDeltaY == 0)
+  {
+    uint8_t diagDirX2, diagDirY2;
+
+    if (absDeltaX == 0)
+    {
+      // Handle edge case where start position is on the edge of the board, so you must move inward
+      diagDirX = (startCol == (TOTAL_UNITSPACES - 1)) ? -1 : 1;
+
+      subDeltaX = 0;
+      subDeltaY = deltaY - (2 * diagDirY);  
+      diagDirX2 = -diagDirX;
+    }
+    else
+    {
+      // Handle edge case where start position is on the edge of the board, so you must move inward
+      diagDirY = (startRow == (TOTAL_UNITSPACES - 1)) ? -1 : 1;
+
+      subDeltaX = deltaX - (2 * diagDirX);
+      subDeltaY = 0; 
+      diagDirY2 = -diagDirY;
+    }
+
+    // Add diagonal movement
+    cols[1] = cols[0] + diagDirX;
+    rows[1] = rows[0] + diagDirY;
+
+    // Add straight movement
+    cols[2] = cols[1] + subDeltaX;
+    rows[2] = rows[1] + subDeltaY;
+
+    // Add diagonal movement, with flipped direction
+    cols[3] = cols[2] + diagDirX2;
+    rows[3] = rows[2] + diagDirY2;
+
+    // 3 added points, plus the initial point
+    numPoints = 4;
+  }
+  // Case where we're moving a cached piece to the graveyard from a capture
+  else if (startCol % 2 == 1  ||  startRow % 2 == 1)
+  {
+    subDeltaX = deltaX - diagDirX; 
+    subDeltaY = deltaY - diagDirY;
```

Comment:
These are only used once, so can just do them in line if you prefer:
```c
cols[1] = cols[0] + (deltaX - diagDirX);
```
and similar for `rows[2]`

Reply:
We could do this, but I would like to keep them the way they are since it makes it a bit more readable. (All of the other array assignments are like this)